### DISCUSSION TITLE
[macOS] [App Store] Are Java apps allowed in the MacOS store?

### DISCUSSION THREAD

**Discussion by agf2013** [shannah/jdeploy#118] [General]
Hello, are apps that use Java currently allowed in MacOS App store?

here is the exact verbiage in their guideline , Java is specifically mentioned as a deprecated technology.
2.4.5 https://developer.apple.com/app-store/review/guidelines/#performance
(viii) Apps should run on the currently shipping OS and may not use deprecated or optionally installed technologies (e.g. Java)

Has Java use been entirely "deprecated" from App Store per guideline? 
Can Jdeploy successfully build .app bundles that will be approved by the App Store review team?  Live examples would help.

Thank you

**Comment by shannah** [comment]
That deprecation notice refers to the old Apple-provided JDK that shipped with Mac OS X 10.0 to 10.6.  It was deprecated the same day that they announced the app store.  Modern Java applications would not rely on this JRE, but rather would bundle a JRE with the app.  jDeploy downloads an appropriate JRE at installation time.   jpackage and jlink are the standard tools that you would use if you want to distribute to the app store.  

They bundle the JRE with your app installer.

If you're not targeting the app store, then jDeploy is much easier and comes out of the box with nice features like auto-update.  If you're targeting the appstore, then, jpackage/jlink.
